src/ralph/schemas/edx/base.py

```diff
@@ -0,0 +1,170 @@
+
+
+class BaseContextSchema(Schema):
+    """Represents the Base Context inherited by all event contexts"""
+
+    course_user_tags = Dict(keys=Str(), values=Str())
+    user_id = Field(required=True, allow_none=True)
+    org_id = Str(required=True)
+    course_id = Str(required=True)
+    path = Url(required=True, relative=True)
+
+    @validates("user_id")
+    def validate_user_id(self, value):
+        """"Check user_id field is None, an empty string or an integer"""
+
+        if value is None or value == "" or isinstance(value, int):
+            return
+        raise ValidationError("user_id should be None, an empty string or an integer")
+
+    @validates_schema
+    def validate_course_id(self, data, **kwargs):
+        """The course_id should be equal to
+        "course-v1:{org_id}+{any_string}+{any_string}"
+        or be an empty string if org_id is an empty string
+        """
+
+        org_id = data["org_id"]
+        course_id = data["course_id"]
+        if not org_id and course_id:
+            raise ValidationError("course_id should be empty if org_id is empty")
+        if org_id and not course_id:
+            raise ValidationError("org_id should be empty if course_id is empty")
+        if not org_id:
+            return
+        if not course_id.startswith("course-v1:"):
+            raise ValidationError("course_id should starts with 'course-v1'")
+        try:
+            organization, course, session = course_id.replace("course-v1:", "").split(
+                "+"
+            )
+        except ValueError as value_error:
+            raise ValidationError(
+                "course_id should contain an organization ID, "
+                "a course name and session separated by a +"
+            ) from value_error
+        if organization != data["org_id"]:
+            raise ValidationError(
+                "organization ID in the course ID does not match the event organization ID"
+            )
+        if len(course) == 0 or len(session) == 0:
+            raise ValidationError("course and session should not be empty")
+
+
+class ContextModuleSchema(Schema):
+    """Represents the context module field"""
+
+    usage_key = Str(required=True)
+    display_name = Str(required=True)
+
+
+class ContextSchema(BaseContextSchema):
+    """Context with module field. Present in Capa problems related
+    events.
+    """
+
+    module = Nested(ContextModuleSchema(), required=True)
+
+    @validates_schema
+    def validate_path(self, data, **kwargs):
+        """Path should start with:
+        "/courses/{course_id}/xblock/block-v1:{course_id[10:]}
+        +type@problem+block@{usage_key}/handler/"
+        """
+
+        path = data["path"]
+        valid_path = (
+            f"/courses/"
+            f"{data['course_id']}"
+            f"/xblock/"
+            f"{data['module']['usage_key']}/"
+            f"handler/"
+        )
+        if path[: len(valid_path)] != valid_path:
```

Comment:
Good point) `not path.startswith(valid_path)` is much better ! Thanks)

Here we check only the start of the {context>path} because we know that the context has a module field so the {context>path} should match something like:

`/courses/{context>course_id}/xblock/{context>module>usage_id}/handler/{hanlder_name}{?/suffix}`

But at this point we don't know what the handler_name is (Example: "xmodule_handler" / "peer_assess") and if there is a suffix (Example: "problem_check" / "hint_button")